samples/demos/ble_xbee_beacon_scan/main.py

```diff
@@ -0,0 +1,218 @@
+# Copyright (c) 2024, Digi International, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import time
+from digi import ble
+
+
+# Constants
+COMPANY_ID_DIGI = bytearray((0xDB, 0x02))
+COMPANY_ID_APPLE = bytearray((0x4C, 0x00))
+
+BEACON_ID_IBEACON = bytearray((0x02, 0x15))
+
+DIGI_BEACON_MIN_LENGTH = 7       # 3 for 'Flags' AD
+#                                  2 for 'Mfg Data' AD
+#                                  2 for Company ID
+DIGI_BEACON_MIN_LENGTH_EXT = 19  # 3 for 'Flags' AD
+#                                  2 for 'Name' AD length and tag
+#                                 10 for 'XBeeMobile' name
+#                                  2 for 'Mfg Data' AD
+#                                  2 for Company ID
+DIGI_COMPANY_INDEX = 5
+DIGI_COMPANY_INDEX_EXT = 17
+
+SCAN_MESSAGE_INTERVAL = 5000  # 5 seconds
+
+
+def is_ibeacon(payload):
```

Comment:
Any reason we're not using the `iBeacon` library for things like this?

(I can see that `iBeacon` doesn't have a way to (cleanly) overwrite the company ID, so I get why we don't use it for parsing the "Digi beacon.)

I also think the license notice from https://github.com/digidotcom/xbee-micropython/tree/master/lib/iBeacon#ibeacon-library-see-license-notice applies here, given the use of the term "iBeacon".